builtin_array.go

```diff
@@ -185,7 +186,6 @@ func (r *Runtime) arrayproto_join(call FunctionCall) Value {
 	}
 
 	var buf valueStringBuilder
-
 	element0 := o.self.getIdx(valueInt(0), nil)
 	if element0 != nil && element0 != _undefined && element0 != _null {
 		buf.WriteString(element0.toString())
```

Comment:
just noticed this, it doesn't look like we are checking the first element in the array.